tests/ui_tools/test_popups.py

```diff
@@ -1398,9 +1398,11 @@ def test_stream_info_content__email_copy_text(
         general_stream: Dict[str, Any],
         stream_email_present: bool,
         expected_copy_text: str,
+        mocker: MockerFixture,
     ) -> None:
         if not stream_email_present:
             del general_stream["email_address"]
+            self.controller.model.stream_copy_text.return_value = ""
 
         model = self.controller.model
         stream_id = general_stream["stream_id"]
```

Comment:
Can this be simplified further? The code doesn't access as much of the model data directly now.

Reply:
I might be wrong, but I don't think we can simplify this. The `StreamInfoView` class calls get_stream_email_address in init, and unless we specify the endpoint return value as `None`, I don't think we can effectively test the possibility of no email being present.

        stream_info_view = StreamInfoView(self.controller, stream_id)

Reply:
Coming back to this, I was referring to the model data, ie. whether we need to have the `del` statement there - it should not depend on the email_address in the stream.

We could rewrite this test to ensure this, but the rest of the function directly accesses the stream_dict for now, so this can likely wait until it gets tidied-away as per another comment.